sips/sip-key-server-temp.md

```diff
@@ -0,0 +1,67 @@
+| SIP-Number          | <Leave this blank; it will be assigned by a SIP Editor> |
+| ---:                | :--- |
+| Title               | BLS-12381 Encryption Key Management for Non Private Key Wallet |
+| Description         | Specify the key derivation logic for an encryption key server |
+| Author              | Joy Wang (@joyqvq, joy@mystenlabs.com)|
+| Editor              | <Leave this blank; it will be assigned by a SIP Editor> |
+| Type                | Standard |
+| Category            | Wallet |
+| Created             | 2024-05-09 |
+| Comments-URI        | <Leave this blank; it will be assigned by a SIP Editor> |
+| Status              | <Leave this blank; it will be assigned by a SIP Editor> |
+| Requires            | <Optional; SIP number(s), comma separated> |
+
+## Abstract
+
+While it is proposed in SIP-TODO for a derivation path from a master private key, there are many use cases that a master private key is not available, such as zkLogin wallet and Multisig wallet. Here we propose a key server example on how an encryption key can be derived from a server master encryption seed based on a unique user identifier as an alternative custodial solution to encryption key management. 
+
+## Motivation
+
+This SIP is proposed as a custodial encryption key derivation solution to address the gaps for SIP-TODO where the master private key is not available. 
+
+## Specification
+
+In many cases there is no available key material in the wallet to derive the encryption key. For example, [zkLogin wallet](https://docs.sui.io/concepts/cryptography/zklogin) does not store any persistent private key to be used for derivation. Multisig wallet also does not have a master private key since the signing operation happens within each individual wallet. 
+
+Here we define a recommended setup for a centralized key server with authentication that can derive and serve a persistent encryption key for each user. 
+
+1. Server holds a master seed. 
+2. Server defines an endpoint and an authentication scheme. The request contains an unique user ID and an application ID.
+3. The server validates and authenticate the request, then responds with an encryption key by deriving it from master seed with domain separator as `sub || app_id` using [HKDF](https://datatracker.ietf.org/doc/html/rfc5869). 
+
```

Comment:
HKDF returns a random byte array - Let's say that the output of the HKDF is of length field size + 128 bit -> and the secret key is computed using modulo the field size.